pkg/asset/installconfig/gcp/validation.go

```diff
@@ -141,66 +141,144 @@ func validateZoneProjects(client API, ic *types.InstallConfig, fieldPath *field.
+				errMsg := fmt.Sprintf("publicDNSZone does not exist in baseDomain %s", ic.BaseDomain)
+				allErrs = append(allErrs, field.Invalid(fieldPath.Child("publicDNSZone").Child("id"), ic.Platform.GCP.PublicDNSZone.ID, errMsg))
+			}
 		}
 	}
 
 	if ic.GCP.PrivateDNSZone != nil && ic.GCP.PrivateDNSZone.ID != "" {
-		if err := commonZoneLookup(ic.GCP.PrivateDNSZone.ID, ic.GCP.PrivateDNSZone.ProjectID, ic.GCP.ProjectID); err != nil {
-			allErrs = append(allErrs, field.Invalid(fieldPath.Child("PrivateDNSZone").Child("ID"), ic.GCP.PrivateDNSZone.ID, "invalid private managed zone"))
+		returnedZone, project, err := internalFindDNSZone(client, ic.Platform.GCP.PrivateDNSZone, ic.GCP.ProjectID)
+		if err != nil {
+			switch {
+			case IsNotFound(err):
+				allErrs = append(allErrs, field.NotFound(field.NewPath("clusterDomain"), fmt.Sprintf("DNS Zone (%s/%s)", project, ic.ClusterDomain())))
+			case IsForbidden(err):
+				errMsg := fmt.Sprintf("Permission denied. Unable to fetch private dns zone information: %s", ic.ClusterDomain())
+				allErrs = append(allErrs, field.Invalid(fieldPath.Child("privateDNSZone").Child("id"), ic.GCP.PrivateDNSZone.ID, errMsg))
+			default:
+				allErrs = append(allErrs, field.InternalError(field.NewPath("clusterDomain"), err))
+			}
+		} else {
+			if !strings.EqualFold(strings.TrimSuffix(returnedZone.DnsName, "."), strings.TrimSuffix(ic.ClusterDomain(), ".")) {
+				errMsg := fmt.Sprintf("dns zone %s did not match expected %s", returnedZone.DnsName, ic.ClusterDomain())
+				allErrs = append(allErrs, field.Invalid(fieldPath.Child("privateManagedZone"), ic.GCP.PrivateDNSZone.ID, errMsg))
+			}
 		}
 	}
 
 	return allErrs
 }
 
-// ValidatePreExitingPublicDNS ensure no pre-existing DNS record exists in the public
-// DNS zone for cluster's Kubernetes API.
-func ValidatePreExitingPublicDNS(client API, ic *types.InstallConfig) error {
+// ValidatePreExistingPrivateDNS ensure that the PrivateZone exists in the cluster domain
+func ValidatePreExistingPrivateDNS(client API, ic *types.InstallConfig) error {
+	record := fmt.Sprintf("api.%s.", strings.TrimSuffix(ic.ClusterDomain(), "."))
+	project := internalFindProject(ic.GCP.PrivateDNSZone, ic.GCP.ProjectID)
+
+	if ic.GCP.PrivateDNSZone != nil && ic.GCP.PrivateDNSZone.ID != "" {
+		zone, err := client.GetDNSZoneByName(context.TODO(), project, ic.GCP.PrivateDNSZone.ID)
+		if err != nil {
+			return fmt.Errorf("failed to find public zone in project %s", project)
+		}
+
+		rrSets, err := client.GetRecordSets(context.TODO(), project, zone.Name)
+		if err != nil {
+			return field.InternalError(field.NewPath("baseDomain"), err)
+		}
+
+		for _, r := range rrSets {
+			if strings.EqualFold(r.Name, record) {
+				errMsg := fmt.Sprintf("record %s already exists in DNS Zone (%s/%s) and might be in use by another cluster, please remove it to continue", record, project, zone.Name)
+				return field.Invalid(field.NewPath("metadata", "name"), ic.ObjectMeta.Name, errMsg)
+			}
+		}
```

Comment:
This is only checking for the api record... if api-int or *.apps exist, that could be problematic. I think it is fair to require it to be empty. I believe that is what AWS does:
https://github.com/openshift/installer/blob/ac9f1c19c96ebab17e67490cf8ccf6504166a566/pkg/asset/installconfig/aws/route53.go#L51-L69

EDIT: Although it looks like this may be using existing code, so perhaps this is "good enough."